plugins/modules/pagerduty_alert.py

```diff
@@ -53,40 +59,89 @@
             - 'triggered'
             - 'acknowledged'
             - 'resolved'
-    api_key:
+    api_version:
         type: str
         description:
-            - The pagerduty API key (readonly access), generated on the pagerduty site.
-        required: true
+            - The API version we want to use to run the module.
+            - V1 is more limited with option we can provide to trigger incident.
+            - V2 has more variables for example, I(severity), I(source), I(custom_detail),etc.
+        default: 'v1'
+        choices:
+            - 'v1'
+            - 'v2'
+        version_added: 6.6.0
+    client:
+        type: str
+        description:
+        - The name of the monitoring client that is triggering this event.
+        required: false
+    client_url:
+        type: str
+        description:
+        -  The URL of the monitoring client that is triggering this event.
+        required: false
+    component:
+        type: str
+        description:
+        -  Component of the source machine that is responsible for the event, for example C(mysql) or C(eth0).
+        required: false
+    custom_details:
+        type: dict
+        description:
+        - Additional details about the event and affected system
+        - A dictionary with custom keys and values.
+        required: false
     desc:
         type: str
         description:
-            - For O(state=triggered) - Required. Short description of the problem that led to this trigger. This field (or a truncated version)
+            - For C(triggered) I(state) - Required. Short description of the problem that led to this trigger. This field (or a truncated version)
```

Comment:
Please keep semantic markup.
```suggestion
            - For O(state=triggered) - Required. Short description of the problem that led to this trigger. This field (or a truncated version)
```